extensions/downloads-manager/src/utils.tsx

```diff
@@ -1,19 +1,86 @@
-import { Action, ActionPanel, confirmAlert, Detail, getPreferenceValues, showToast, Toast, trash } from "@raycast/api";
+import {
+  Action,
+  ActionPanel,
+  Cache,
+  confirmAlert,
+  Detail,
+  getPreferenceValues,
+  showToast,
+  Toast,
+  trash,
+} from "@raycast/api";
 import { showFailureToast } from "@raycast/utils";
 import { accessSync, constants, readdirSync, statSync } from "fs";
 import { rm } from "fs/promises";
 import { join } from "path";
+import { execSync } from "child_process";
+import { homedir } from "os";
 import { ComponentType } from "react";
 import untildify from "untildify";
 
+const cache = new Cache();
+const CACHE_KEY = "customDownloadsFolder";
+
 const preferences = getPreferenceValues();
-export const downloadsFolder = untildify(preferences.downloadsFolder ?? "~/Downloads");
+
+function getCachedOrDetectDownloadsFolder(): string {
+  // If preference is set, use it
+  if (preferences.downloadsFolder && preferences.downloadsFolder.trim()) {
+    return untildify(preferences.downloadsFolder);
+  }
+
+  // Check cache first
+  const cached = cache.get(CACHE_KEY);
+  if (cached && typeof cached === "string") {
+    return cached;
+  }
+
+  // Detect and cache the folder
+  const detected = getCustomDownloadsFolder();
+  cache.set(CACHE_KEY, detected);
+  return detected;
+}
+
+export const downloadsFolder = getCachedOrDetectDownloadsFolder();
 const showHiddenFiles = preferences.showHiddenFiles;
 const fileOrder = preferences.fileOrder;
-const lastestDownloadOrder = preferences.lastestDownloadOrder;
+const latestDownloadOrder = preferences.lastestDownloadOrder;
 export const defaultDownloadsLayout = preferences.downloadsLayout ?? "list";
 const imageExtensions = [".png", ".jpg", ".jpeg", ".gif", ".webp", ".bmp", ".tiff", ".heic", ".svg"];
 
+export function getCustomDownloadsFolder(): string {
+  // macOS
+  if (process.platform === "darwin") {
+    // Query Finder for the actual Downloads folder location
+    try {
+      const result = execSync(
+        `osascript -e 'tell application "Finder" to return POSIX path of (path to downloads folder)'`,
+        { encoding: "utf-8" },
+      );
+      return result.trim();
+    } catch (error) {
+      // Fallback to default location if query fails
+      console.error("Failed to get Downloads folder from Finder:", error);
+      return untildify("~/Downloads");
+    }
```

Comment:
```suggestion
    return untildify("~/Downloads");
```

let's just use `~/Downloads` on macos, there are no other path